kratos/sources/kratos_parameters.cpp

```diff
@@ -629,6 +629,38 @@ Matrix Parameters::GetMatrix() const
 /***********************************************************************************/
 /***********************************************************************************/
 
+#define KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(TYPE, TYPE_NAME)                                \
+    template <> bool Parameters::Is<TYPE>() const {return this->Is ## TYPE_NAME();}              \
+    template <> TYPE Parameters::Get<TYPE>() const {return this->Get ## TYPE_NAME();}            \
+    template <> void Parameters::Set<TYPE>(const TYPE& rValue) {this->Set ## TYPE_NAME(rValue);}
+
+KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(double, Double)
+
+KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(int, Int)
+
+KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(bool, Bool)
+
+KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(std::string, String)
+
+//KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(std::vector<std::string>, StringArray) // <== missing Parameters::IsStringArray()
+
+KRATOS_DEFINE_PARAMETERS_VALUE_ACCESSORS(Vector, Vector)
```

Comment:
in my opinion this makes no sense since Vector and Matrix are not json entities...

Reply:
all the templated```Is```, ```Set```, and ```Get``` functions just reroute to their already implemented counterparts. In the case of ```Matrix```:
- ```Is<Matrix>()``` invokes ```IsMatrix()```
- ```Get<Matrix>()``` invokes ```GetMatrix()```
- ```Set<Matrix>(const Matrix&)``` invokes ```Set(const Matrix&)```
so they only make as much sense as the functions we already have

Reply:
ping @RiccardoRossi